dask_geopandas/morton_distance.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import pandas as pd
+from dask_geopandas.hilbert_distance import _continuous_to_discrete_coords
+
+# Based on #https://code.activestate.com/recipes/
+# 577558-interleave-bits-aka-morton-ize-aka-z-order-curve/
+
+
+def _morton_distance(gdf, total_bounds, p):
+    """
+    Calculate
+
+    Parameters
+    ----------
+
+    gdf : GeoDataFrame
+
+    total_bounds : Total bounds
+
+    p : number of iterations
+
+    """
+    bounds = gdf.bounds.to_numpy()
+    coords = _continuous_to_discrete_coords(total_bounds, bounds, p)
+
+    distances = _distances_from_coordinates(coords)
+
+    return pd.Series(distances, index=gdf.index, name="morton_distance")
+
+
+def _distances_from_coordinates(coords):
+
+    """
+    Calculate Morton distances from coordinates
+
+    Parameters
+    ----------
+
+    coords : Coordinates in np.array format
+    """
+
+    result = np.zeros(coords.shape[0], dtype=np.int64)
+    # For each coord calculate hilbert distance
+    for i in range(coords.shape[0]):
+        result[i] = encode_morton(coords[i][0], coords[i][1])
```

Comment:
```suggestion
    result = encode_morton(coords[:, 0], coords[:, 1])
```

(see comment below about this (probably) working on full numpy arrays as well)